sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/auth/DeviceAuthenticator.kt

```diff
@@ -92,4 +103,45 @@ class RealDeviceAuthenticator @Inject constructor(
             onResult(AuthResult.Success)
         }
     }
+
+    @SuppressLint("InlinedApi", "DEPRECATION")
+    override fun launchDeviceAuthEnrollment(context: Context) {
+        when {
+            appBuildConfig.manufacturer == "Xiaomi" -> {
+                // Issue on Xiaomi: https://stackoverflow.com/questions/68484485/intent-action-fingerprint-enroll-on-redmi-results-in-exception
+                SYSTEM_SETTINGS_ACTION.safeLaunchSettingsActivity(context, tryFallback = false)
+            }
+
+            appBuildConfig.sdkInt >= Build.VERSION_CODES.R -> {
+                Settings.ACTION_BIOMETRIC_ENROLL.safeLaunchSettingsActivity(context, tryFallback = true)
+            }
+
+            appBuildConfig.sdkInt >= Build.VERSION_CODES.P -> {
+                Settings.ACTION_FINGERPRINT_ENROLL.safeLaunchSettingsActivity(context, tryFallback = true)
+            }
+
+            else -> {
+                Settings.ACTION_SECURITY_SETTINGS.safeLaunchSettingsActivity(context, tryFallback = true)
+            }
+        }
+    }
+
```

Comment:
Duplicating this between autofill and sync (and whichever other feature needs all of this) isn't a great strategy long term. It's _ok_ for now probably, but we should start identifying which parts are truly common and pulling them out for re-use vs duplicating.

Reply:
taking note of this, agree.